server/commands/write.go

```diff
@@ -60,9 +60,37 @@ func (c *WriteCommand) validateTuplesets(ctx context.Context, req *openfgapb.Wri
 	}
 
 	for _, tk := range writes {
-		if _, err := tupleUtils.ValidateTuple(ctx, c.datastore, store, modelID, tk, dbCallsCounter); err != nil {
+		tupleUserset, err := tupleUtils.ValidateTuple(ctx, c.datastore, store, modelID, tk, dbCallsCounter)
+		if err != nil {
 			return serverErrors.HandleTupleValidateError(err)
 		}
+		switch usType := tupleUserset.Userset.(type) {
```

Comment:
Consider putting this on its own function.

Reply:
@elbuo8 - Thank you for this, I like the idea and have implemented this in the latest commit.  It definitely increased readability and, if ever needed, gives a function for assessing direct/indirect relations for future use